src/parquet/column/column-writer-test.cc

```diff
@@ -334,5 +345,28 @@ TYPED_TEST(TestPrimitiveWriter, RequiredLargeChunk) {
   ASSERT_EQ(this->values_, this->values_out_);
 }
 
+// Test case for dictionary fallback encoding skip BooleanType
+TYPED_TEST(TestPrimitiveWriter, RequiredVeryLargeChunk) {
+  this->GenerateData(VERY_LARGE_SIZE);
+
+  auto writer = this->BuildWriter(VERY_LARGE_SIZE, Encoding::PLAIN_DICTIONARY);
+  writer->WriteBatch(this->values_.size(), nullptr, nullptr, this->values_ptr_);
+  writer->Close();
+
+  // Just read the first SMALL_SIZE rows to ensure we could read it back in
+  this->ReadColumn();
+  ASSERT_EQ(SMALL_SIZE, this->values_read_);
+  this->values_.resize(SMALL_SIZE);
+  ASSERT_EQ(this->values_, this->values_out_);
+  if (this->type_num() != Type::BOOLEAN) {
+    // There are 3 encodings (RLE, PLAIN_DICTIONARY, PLAIN) in a fallback case
+    ASSERT_EQ(3, this->metadata_num_encodings());
```

Comment:
At this risk of being pedantic, I wonder if we should explicitly check that the encodings vector contains either `PLAIN_DICTIONARY` or `RLE_DICTIONARY`